plugins/provisioners/ansible/provisioner/base.rb

```diff
@@ -134,6 +141,12 @@ def generate_inventory_groups
               defined_groups << gname
               inventory_groups += "\n[#{gname}]\n"
               gmembers.each do |gm|
+              # TODO : Expand and validate host range patterns
+              # against @inventory_machines list before adding them
+              # otherwise abort with an error message
+                if gm[/(?:\[[a-z]:[a-z]\]|\[[0-9]+?:[0-9]+?\])/]
```

Comment:
Store the regexp in a [freezed constant](https://github.com/mitchellh/vagrant/blob/master/plugins/provisioners/ansible/config/base.rb#L6) (used twice in this file)